eta-company-calendar-backend/src/main/java/hu/flowacademy/companycalendar/config/CustomTokenEnhancer.java

```diff
@@ -0,0 +1,33 @@
+package hu.flowacademy.companycalendar.config;
+
+import hu.flowacademy.companycalendar.model.User;
+import hu.flowacademy.companycalendar.repository.UserRepository;
+import java.util.HashMap;
+import java.util.Map;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.oauth2.common.DefaultOAuth2AccessToken;
+import org.springframework.security.oauth2.common.OAuth2AccessToken;
+import org.springframework.security.oauth2.provider.OAuth2Authentication;
+import org.springframework.security.oauth2.provider.token.TokenEnhancer;
+
+@Slf4j
+public class CustomTokenEnhancer implements TokenEnhancer {
+
+  @Autowired
+  private UserRepository userRepository;
+
+  @Override
+  public OAuth2AccessToken enhance(OAuth2AccessToken token,
+      OAuth2Authentication auth) {
+    Object principal = auth.getPrincipal();
+    if (principal instanceof User) {
+      var user = (User) principal;
+      ((DefaultOAuth2AccessToken) token).setAdditionalInformation(
+          Map.of("id", user.getId()));
+      return token;
+    }
+    log.error("Principal is not User.class");
```

Comment:
Hmm. És az nem probléma? Mármint, hogy így is visszatérünk a `token`-nel

Reply:
Elvileg nem gond.

@borosr

Reply:
Szerintem ide nagyjából csak akkor kerülhet, ha nincs bejelentkezve a felhasználó, de hozzáfér egy oldalhoz. Olyankor, ha jól emlékszem a principal típusa string lesz, az értéke pedig "annonymususer"